_src/js/compare/index.jsx

```diff
@@ -86,17 +94,25 @@ class Compare extends React.Component {
   }
 
   selectBudget1 (option) {
-    this.selectBudget('budget1', option.value);
+    this.selectBudget('budget1', 'budget2', option.value);
   }
 
   selectBudget2 (option) {
-    this.selectBudget('budget2', option.value);
+    this.selectBudget('budget2', 'budget1', option.value);
   }
 
-  selectBudget (key, index) {
+  selectBudget (key, otherKey, index) {
+    // No change if same selection
+    if (this.state[`${key}Choice`] === index) {
+      return;
+    }
+
+    const budgetOptions = this.state.budgetChoices.slice();
```

Comment:
might be good to call this `otherBudgetOptions` or something like that, for clarity